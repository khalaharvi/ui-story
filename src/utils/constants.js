export const PRIMARY_CATEGORY_KEY = 'categories.lvl0';
export const SECONDARY_CATEGORY_KEY = 'categories.lvl1';
export const TERTIARY_CATEGORY_KEY = 'categories.lvl2';
