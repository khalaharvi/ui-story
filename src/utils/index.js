export { default as createClassNames } from './createClassNames';
export { default as getDisplayName } from './getDisplayName';
export { default as capitalizeFirstLetter } from './capitalizeFirstLetter';
export { default as slugify } from './slugify';
