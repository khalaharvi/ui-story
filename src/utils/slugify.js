export default (text) => {
  // Use hash map for special characters
  const specialChars = {"à":'a',"ä":'a',"á":'a',"â":'a',"æ":'a',"å":'a',"ë":'e',"è":'e',"é":'e', "ê":'e',"î":'i',"ï":'i',"ì":'i',"í":'i',"ò":'o',"ó":'o',"ö":'o',"ô":'o',"ø":'o',"ù":'o',"ú":'u',"ü":'u',"û":'u',"ñ":'n',"ç":'c',"ß":'s',"ÿ":'y',"œ":'o',"ŕ":'r',"ś":'s',"ń":'n',"ṕ":'p',"ẃ":'w',"ǵ":'g',"ǹ":'n',"ḿ":'m',"ǘ":'u',"ẍ":'x',"ź":'z',"ḧ":'h',"·":'-',"/":'-',"_":'-',",":'-',":":'-',";":'-'}; // eslint-disable-line

  return text.toString().toLowerCase()
    .replace(/\s+/g, '-') // Replace spaces with -
    .replace(/./g, target => specialChars[target] || target) // Replace special characters using the hash map
    .replace(/&/g, '-and-') // Replace & with 'and'
    .replace(/[^\w-]+/g, '') // Remove all non-word chars
    .replace(/--+/g, '-') // Replace multiple - with single -
    .replace(/^-+/, '') // Trim - from start of text
    .replace(/-+$/, ''); // Trim - from end of text
};
