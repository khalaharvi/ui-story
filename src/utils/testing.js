import { object } from 'prop-types';
import { channel, createBroadcast } from 'emotion-theming';
import { shallow, mount } from 'enzyme';
import theme from '@dering-hall/dh-theme';

const getOptions = (options = {}) => {
  const broadcast = createBroadcast(theme);

  return {
    context: {
      [channel]: broadcast
    },
    childContextTypes: {
      [channel]: object
    },
    ...options
  };
};

export const mountWithTheme = (
  Component,
  options = {}
) => mount(Component, { ...getOptions(options) });

export const shallowWithTheme = (
  Component,
  options = {}
) => shallow(Component, { ...getOptions(options) });

export default {
  mountWithTheme,
  shallowWithTheme
};
