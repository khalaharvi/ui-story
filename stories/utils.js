/* eslint-disable import/prefer-default-export */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { InstantSearch } from 'react-instantsearch/dom';
import { connectStateResults } from 'react-instantsearch/connectors';
import { withMedia, MediaQueryProvider } from 'react-media-query-hoc';
import { withTests as withTestsAddon } from '@storybook/addon-jest';
import { ThemeProvider } from 'emotion-theming';
import dhTheme from '@dering-hall/dh-theme';
import testsResults from '../.jest-test-results.json';

const Wrap = ({ index, children }) => (
  <InstantSearch
    appId={process.env.ALGOLIA_ID}
    apiKey={process.env.ALGOLIA_API_KEY}
    indexName={index}
  >
    {children}
  </InstantSearch>
);

Wrap.propTypes = {
  index: PropTypes.string,
  children: PropTypes.node
};

Wrap.defaultProps = {
  index: process.env.ALGOLIA_PRODUCT_INDEX
};

const WrapThemeProvider = ({ children, theme }) => (
  <ThemeProvider theme={theme}>
    <Fragment>
      {children}
    </Fragment>
  </ThemeProvider>
);

WrapThemeProvider.propTypes = {
  theme: PropTypes.object // eslint-disable-line
};

WrapThemeProvider.defaultProps = {
  theme: dhTheme
};

const SearchStateDebug = connectStateResults(({ searchState }) => (
  <div style={{ margin: '1rem 0', fontFamily: 'monospace' }}>
    <pre
      style={{
        background: '#f6f8fa',
        fontSize: '.65rem',
        padding: '.5rem'
      }}
    >
      <strong>searchState</strong> = {JSON.stringify(searchState, null, '\t')}
    </pre>
  </div>
));

const customQueries = {
  small: 'screen and (max-width: 730px)',
  medium: 'screen and (max-width: 970px)',
  large: 'screen and (max-width: 1240px)',
  xlarge: 'screen and (min-width: 1240px)'
};

const MediaQueriesWrapper = ({ children }) => (
  <MediaQueryProvider queries={customQueries}>
    {children}
  </MediaQueryProvider>
);

const OnSmallOnly = withMedia(({ media, children }) => {
  if (!media.small) {
    return null;
  }

  return (
    <Fragment>
      {children}
    </Fragment>
  );
});

const withTests = withTestsAddon({
  results: testsResults
});

export {
  SearchStateDebug,
  MediaQueriesWrapper,
  OnSmallOnly,
  Wrap,
  WrapThemeProvider,
  withTests
};
/* eslint-enable */
