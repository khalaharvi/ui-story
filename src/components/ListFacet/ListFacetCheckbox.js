import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _pick from 'lodash.pick';
import { connectRefinementList } from 'react-instantsearch/connectors';

import { createClassNames } from 'utils';
import PanelCallbackHandler from 'components/Panel/PanelCallbackHandler';
import ListFacetOptions from './ListFacetOptions';
import ListFacetCheckboxLabel from './ListFacetCheckboxLabel';

const cx = createClassNames('list-facet');

const itemsPropType = PropTypes.arrayOf(
  PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string,
    count: PropTypes.number.isRequired,
    items: (...args) => itemsPropType(...args)
  })
);

export class ListFacetCheckbox extends Component {
  static propTypes = {
    refine: PropTypes.func.isRequired,
    canRefine: PropTypes.bool.isRequired,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.arrayOf(PropTypes.string).isRequired,
        count: PropTypes.number.isRequired,
        isRefined: PropTypes.bool.isRequired
      })
    ),
    showMore: PropTypes.bool,
    limit: PropTypes.number,
    showMoreLimit: PropTypes.number,
    transformItems: PropTypes.func,
    className: PropTypes.string
  };

  renderItem = (item) => {
    const { refine } = this.props;

    return (
      <ListFacetCheckboxLabel className={cx('option--checkbox')} isRefined={item.isRefined}>
        <input
          type="checkbox"
          checked={item.isRefined}
          onChange={() => refine(item.value)}
        />
        <span className={cx('option-label')}>{item.label}</span>
      </ListFacetCheckboxLabel>
    );
  };

  render() {
    return (
      <ListFacetOptions
        renderItem={this.renderItem}
        cx={cx}
        {..._pick(this.props, [
          'items',
          'showMore',
          'limit',
          'showMoreLimit',
          'canRefine',
          'className'
        ])}
      />
    );
  }
}

const ListFacetCheckboxComponent = props => (
  <PanelCallbackHandler {...props}>
    <ListFacetCheckbox {...props} />
  </PanelCallbackHandler>
);

export default connectRefinementList(ListFacetCheckboxComponent);
