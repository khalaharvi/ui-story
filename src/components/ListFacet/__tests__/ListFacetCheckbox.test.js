import React from 'react';
import { mountWithTheme } from 'utils/testing';

import { ListFacetCheckbox } from '../ListFacetCheckbox';

describe('<ListFacetCheckbox />', () => {
  it('renders correctly', () => {
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={() => null}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with className', () => {
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={() => null}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
        className="custom-list-facet-checkbox"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={() => null}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('refines its value on change', () => {
    const refine = jest.fn();
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={refine}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          }
        ]}
        canRefine={false}
      />
    );

    const items = wrapper.find('li');

    expect(items).toHaveLength(3);

    const firstItem = items.first().find('input[type="checkbox"]');

    firstItem.simulate('change', { target: { checked: true } });

    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual(['white']);

    wrapper.unmount();
  });

  it('should respect defined limits', () => {
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={() => null}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          },
          {
            label: 'red',
            value: ['red'],
            count: 30,
            isRefined: false
          },
          {
            label: 'green',
            value: ['green'],
            count: 30,
            isRefined: false
          }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );

    const items = wrapper.find('li');

    expect(items).toHaveLength(2);

    wrapper.find('button').simulate('click');

    expect(wrapper.find('li')).toHaveLength(4);

    wrapper.unmount();
  });

  it('should disable show more when no more item to display', () => {
    const wrapper = mountWithTheme(
      <ListFacetCheckbox
        refine={() => null}
        items={[
          {
            label: 'white',
            value: ['white'],
            count: 10,
            isRefined: true
          },
          {
            label: 'black',
            value: ['black'],
            count: 20,
            isRefined: false
          },
          {
            label: 'blue',
            value: ['blue'],
            count: 30,
            isRefined: false
          }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );

    const items = wrapper.find('li');

    expect(items).toHaveLength(2);

    expect(wrapper.find('button[disabled]')).toBeDefined();

    wrapper.unmount();
  });
});
