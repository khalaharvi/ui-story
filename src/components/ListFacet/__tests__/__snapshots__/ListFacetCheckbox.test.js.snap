// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFacetCheckbox /> renders correctly 1`] = `
.emotion-12 {
  position: relative;
}

.emotion-8 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.emotion-8 ul {
  padding-left: 15px;
  margin-bottom: 0;
}

.emotion-2 {
  display: block;
  position: relative;
  color: #242424;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-2:hover {
  color: #242424;
}

.emotion-2 > a {
  color: #242424;
}

.emotion-2 > a + ul {
  display: block;
}

.emotion-0 {
  position: relative;
  cursor: pointer;
  color: #242424;
  display: block;
}

.emotion-0:hover {
  color: #242424;
}

.emotion-0 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-0 span {
  display: block;
}

.emotion-6 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-6:hover {
  color: #242424;
}

.emotion-4 {
  position: relative;
  cursor: pointer;
  color: #6d6d6a;
  display: block;
}

.emotion-4:hover {
  color: #242424;
}

.emotion-4 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-4 span {
  display: block;
}

.emotion-10 {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: none;
  cursor: pointer;
  margin: 0;
  color: #cdcbc5;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-10:hover {
  color: #242424;
}

.emotion-10[disabled] {
  display: none;
}

<ListFacetCheckbox
  canRefine={true}
  items={
    Array [
      Object {
        "count": 10,
        "isRefined": true,
        "label": "white",
        "value": Array [
          "white",
        ],
      },
      Object {
        "count": 20,
        "isRefined": false,
        "label": "black",
        "value": Array [
          "black",
        ],
      },
      Object {
        "count": 30,
        "isRefined": false,
        "label": "blue",
        "value": Array [
          "blue",
        ],
      },
    ]
  }
  limit={2}
  refine={[Function]}
  showMore={true}
  showMoreLimit={4}
>
  <WithTheme(ListFacetOptions)
    canRefine={true}
    cx={[Function]}
    items={
      Array [
        Object {
          "count": 10,
          "isRefined": true,
          "label": "white",
          "value": Array [
            "white",
          ],
        },
        Object {
          "count": 20,
          "isRefined": false,
          "label": "black",
          "value": Array [
            "black",
          ],
        },
        Object {
          "count": 30,
          "isRefined": false,
          "label": "blue",
          "value": Array [
            "blue",
          ],
        },
      ]
    }
    limit={2}
    renderItem={[Function]}
    showMore={true}
    showMoreLimit={4}
  >
    <ListFacetOptions
      canRefine={true}
      cx={[Function]}
      items={
        Array [
          Object {
            "count": 10,
            "isRefined": true,
            "label": "white",
            "value": Array [
              "white",
            ],
          },
          Object {
            "count": 20,
            "isRefined": false,
            "label": "black",
            "value": Array [
              "black",
            ],
          },
          Object {
            "count": 30,
            "isRefined": false,
            "label": "blue",
            "value": Array [
              "blue",
            ],
          },
        ]
      }
      limit={2}
      renderItem={[Function]}
      showMore={true}
      showMoreLimit={4}
    >
      <Styled(div)
        canRefine={true}
        className="dh-list-facet"
      >
        <div
          className="dh-list-facet emotion-12 emotion-13"
        >
          <Styled(ul)
            canRefine={true}
          >
            <ul
              className="emotion-8 emotion-9"
            >
              <Styled(li)
                className="dh-list-facet-option dh-list-facet-option--active"
                isRefined={true}
                key="white"
                parent={false}
              >
                <li
                  className="dh-list-facet-option dh-list-facet-option--active emotion-2 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={true}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-0 emotion-1"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        white
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
              <Styled(li)
                className="dh-list-facet-option"
                isRefined={false}
                key="black"
                parent={false}
              >
                <li
                  className="dh-list-facet-option emotion-6 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={false}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-4 emotion-1"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        black
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
            </ul>
          </Styled(ul)>
          <Styled(button)
            className="dh-list-facet-option--show-more"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="dh-list-facet-option--show-more emotion-10 emotion-11"
              disabled={false}
              onClick={[Function]}
            >
              View 
              More
            </button>
          </Styled(button)>
        </div>
      </Styled(div)>
    </ListFacetOptions>
  </WithTheme(ListFacetOptions)>
</ListFacetCheckbox>
`;

exports[`<ListFacetCheckbox /> renders correctly with className 1`] = `
.emotion-12 {
  position: relative;
}

.emotion-8 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.emotion-8 ul {
  padding-left: 15px;
  margin-bottom: 0;
}

.emotion-2 {
  display: block;
  position: relative;
  color: #242424;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-2:hover {
  color: #242424;
}

.emotion-2 > a {
  color: #242424;
}

.emotion-2 > a + ul {
  display: block;
}

.emotion-0 {
  position: relative;
  cursor: pointer;
  color: #242424;
  display: block;
}

.emotion-0:hover {
  color: #242424;
}

.emotion-0 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-0 span {
  display: block;
}

.emotion-6 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-6:hover {
  color: #242424;
}

.emotion-4 {
  position: relative;
  cursor: pointer;
  color: #6d6d6a;
  display: block;
}

.emotion-4:hover {
  color: #242424;
}

.emotion-4 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-4 span {
  display: block;
}

.emotion-10 {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: none;
  cursor: pointer;
  margin: 0;
  color: #cdcbc5;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-10:hover {
  color: #242424;
}

.emotion-10[disabled] {
  display: none;
}

<ListFacetCheckbox
  canRefine={true}
  className="custom-list-facet-checkbox"
  items={
    Array [
      Object {
        "count": 10,
        "isRefined": true,
        "label": "white",
        "value": Array [
          "white",
        ],
      },
      Object {
        "count": 20,
        "isRefined": false,
        "label": "black",
        "value": Array [
          "black",
        ],
      },
      Object {
        "count": 30,
        "isRefined": false,
        "label": "blue",
        "value": Array [
          "blue",
        ],
      },
    ]
  }
  limit={2}
  refine={[Function]}
  showMore={true}
  showMoreLimit={4}
>
  <WithTheme(ListFacetOptions)
    canRefine={true}
    className="custom-list-facet-checkbox"
    cx={[Function]}
    items={
      Array [
        Object {
          "count": 10,
          "isRefined": true,
          "label": "white",
          "value": Array [
            "white",
          ],
        },
        Object {
          "count": 20,
          "isRefined": false,
          "label": "black",
          "value": Array [
            "black",
          ],
        },
        Object {
          "count": 30,
          "isRefined": false,
          "label": "blue",
          "value": Array [
            "blue",
          ],
        },
      ]
    }
    limit={2}
    renderItem={[Function]}
    showMore={true}
    showMoreLimit={4}
  >
    <ListFacetOptions
      canRefine={true}
      className="custom-list-facet-checkbox"
      cx={[Function]}
      items={
        Array [
          Object {
            "count": 10,
            "isRefined": true,
            "label": "white",
            "value": Array [
              "white",
            ],
          },
          Object {
            "count": 20,
            "isRefined": false,
            "label": "black",
            "value": Array [
              "black",
            ],
          },
          Object {
            "count": 30,
            "isRefined": false,
            "label": "blue",
            "value": Array [
              "blue",
            ],
          },
        ]
      }
      limit={2}
      renderItem={[Function]}
      showMore={true}
      showMoreLimit={4}
    >
      <Styled(div)
        canRefine={true}
        className="dh-list-facet custom-list-facet-checkbox"
      >
        <div
          className="dh-list-facet custom-list-facet-checkbox emotion-12 emotion-13"
        >
          <Styled(ul)
            canRefine={true}
          >
            <ul
              className="emotion-8 emotion-9"
            >
              <Styled(li)
                className="dh-list-facet-option dh-list-facet-option--active"
                isRefined={true}
                key="white"
                parent={false}
              >
                <li
                  className="dh-list-facet-option dh-list-facet-option--active emotion-2 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={true}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-0 emotion-1"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        white
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
              <Styled(li)
                className="dh-list-facet-option"
                isRefined={false}
                key="black"
                parent={false}
              >
                <li
                  className="dh-list-facet-option emotion-6 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={false}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-4 emotion-1"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        black
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
            </ul>
          </Styled(ul)>
          <Styled(button)
            className="dh-list-facet-option--show-more"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="dh-list-facet-option--show-more emotion-10 emotion-11"
              disabled={false}
              onClick={[Function]}
            >
              View 
              More
            </button>
          </Styled(button)>
        </div>
      </Styled(div)>
    </ListFacetOptions>
  </WithTheme(ListFacetOptions)>
</ListFacetCheckbox>
`;

exports[`<ListFacetCheckbox /> renders correctly without refinement 1`] = `
.emotion-12 {
  position: relative;
}

.emotion-8 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.emotion-8 ul {
  padding-left: 15px;
  margin-bottom: 0;
}

.emotion-2 {
  display: block;
  position: relative;
  color: #242424;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-2:hover {
  color: #242424;
}

.emotion-2 > a {
  color: #242424;
}

.emotion-2 > a + ul {
  display: block;
}

.emotion-0 {
  position: relative;
  cursor: pointer;
  color: #242424;
  display: block;
}

.emotion-0:hover {
  color: #242424;
}

.emotion-0 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-0 span {
  display: block;
}

.emotion-6 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-6:hover {
  color: #242424;
}

.emotion-4 {
  position: relative;
  cursor: pointer;
  color: #6d6d6a;
  display: block;
}

.emotion-4:hover {
  color: #242424;
}

.emotion-4 input[type="checkbox"] {
  position: absolute;
  left: -9999px;
}

.emotion-4 span {
  display: block;
}

.emotion-10 {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: none;
  cursor: pointer;
  margin: 0;
  color: #cdcbc5;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-10:hover {
  color: #242424;
}

.emotion-10[disabled] {
  display: none;
}

<ListFacetCheckbox
  canRefine={false}
  items={
    Array [
      Object {
        "count": 10,
        "isRefined": true,
        "label": "white",
        "value": Array [
          "white",
        ],
      },
      Object {
        "count": 20,
        "isRefined": false,
        "label": "black",
        "value": Array [
          "black",
        ],
      },
      Object {
        "count": 30,
        "isRefined": false,
        "label": "blue",
        "value": Array [
          "blue",
        ],
      },
    ]
  }
  limit={2}
  refine={[Function]}
  showMore={true}
  showMoreLimit={4}
>
  <WithTheme(ListFacetOptions)
    canRefine={false}
    cx={[Function]}
    items={
      Array [
        Object {
          "count": 10,
          "isRefined": true,
          "label": "white",
          "value": Array [
            "white",
          ],
        },
        Object {
          "count": 20,
          "isRefined": false,
          "label": "black",
          "value": Array [
            "black",
          ],
        },
        Object {
          "count": 30,
          "isRefined": false,
          "label": "blue",
          "value": Array [
            "blue",
          ],
        },
      ]
    }
    limit={2}
    renderItem={[Function]}
    showMore={true}
    showMoreLimit={4}
  >
    <ListFacetOptions
      canRefine={false}
      cx={[Function]}
      items={
        Array [
          Object {
            "count": 10,
            "isRefined": true,
            "label": "white",
            "value": Array [
              "white",
            ],
          },
          Object {
            "count": 20,
            "isRefined": false,
            "label": "black",
            "value": Array [
              "black",
            ],
          },
          Object {
            "count": 30,
            "isRefined": false,
            "label": "blue",
            "value": Array [
              "blue",
            ],
          },
        ]
      }
      limit={2}
      renderItem={[Function]}
      showMore={true}
      showMoreLimit={4}
    >
      <Styled(div)
        canRefine={false}
        className="dh-list-facet"
      >
        <div
          className="dh-list-facet emotion-12 emotion-13"
        >
          <Styled(ul)
            canRefine={false}
          >
            <ul
              className="emotion-8 emotion-9"
            >
              <Styled(li)
                className="dh-list-facet-option dh-list-facet-option--active"
                isRefined={true}
                key="white"
                parent={false}
              >
                <li
                  className="dh-list-facet-option dh-list-facet-option--active emotion-2 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={true}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-0 emotion-1"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        white
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
              <Styled(li)
                className="dh-list-facet-option"
                isRefined={false}
                key="black"
                parent={false}
              >
                <li
                  className="dh-list-facet-option emotion-6 emotion-3"
                >
                  <Styled(label)
                    className="dh-list-facet-option--checkbox"
                    isRefined={false}
                  >
                    <label
                      className="dh-list-facet-option--checkbox emotion-4 emotion-1"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="dh-list-facet-option-label"
                      >
                        black
                      </span>
                    </label>
                  </Styled(label)>
                </li>
              </Styled(li)>
            </ul>
          </Styled(ul)>
          <Styled(button)
            className="dh-list-facet-option--show-more"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="dh-list-facet-option--show-more emotion-10 emotion-11"
              disabled={false}
              onClick={[Function]}
            >
              View 
              More
            </button>
          </Styled(button)>
        </div>
      </Styled(div)>
    </ListFacetOptions>
  </WithTheme(ListFacetOptions)>
</ListFacetCheckbox>
`;
